public/Aryan Suri/Sidebar/sidebar.js

```diff
@@ -0,0 +1,1645 @@
+}
+
+function FillMenuWithArray(myMenu, myArray) {
+  // Fills the options of myMenu with the elements of myArray.
+  // !CAUTION!: It replaces the elements, so old ones will be deleted.
+  var i;
+  myMenu.length = myArray.length;
+  for (i = 0; i < myArray.length; i++) {
+    myMenu.options[i].text = myArray[i];
+  }
+}
+
+function CalculateUnit(sourceForm, targetForm) {
+  // A simple wrapper function to validate input before making the conversion
+  var sourceValue = sourceForm.unit_input.value;
+
+  // First check if the user has given numbers or anything that can be made to one...
+  sourceValue = parseFloat(sourceValue);
+  if (!isNaN(sourceValue) || sourceValue == 0) {
+    // If we can make a valid floating-point number, put it in the text box and convert!
+    sourceForm.unit_input.value = sourceValue;
+    ConvertFromTo(sourceForm, targetForm);
+  }
+}
+
+function ConvertFromTo(sourceForm, targetForm) {
+  // Converts the contents of the sourceForm input box to the units specified in the targetForm unit menu and puts the result in the targetForm input box.In other words, this is the heart of the whole script...
+  var propIndex;
+  var sourceIndex;
+  var sourceFactor;
+  var targetIndex;
+  var targetFactor;
+  var result;
+
+  // Start by checking which property we are working in...
+  propIndex = document.property_form.the_menu.selectedIndex;
+
+  // Let's determine what unit are we converting FROM (i.e. source) and the factor needed to convert that unit to the base unit.
+  sourceIndex = sourceForm.unit_menu.selectedIndex;
+  sourceFactor = factor[propIndex][sourceIndex];
+
+  // Cool! Let's do the same thing for the target unit - the units we are converting TO:
+  targetIndex = targetForm.unit_menu.selectedIndex;
+  targetFactor = factor[propIndex][targetIndex];
+
+  // Simple, huh? let's do the math: a) convert the source TO the base unit: (The input has been checked by the CalculateUnit function).
+
+  result = sourceForm.unit_input.value;
+  // Handle Temperature increments!
+  if (property[propIndex] == "Temperature") {
+    result = parseFloat(result) + tempIncrement[sourceIndex];
+  }
+  result = result * sourceFactor;
+
+  // not done yet... now, b) use the targetFactor to convert FROM the base unit
+  // to the target unit...
+  result = result / targetFactor;
+  // Again, handle Temperature increments!
+  if (property[propIndex] == "Temperature") {
+    result = parseFloat(result) - tempIncrement[targetIndex];
+  }
+
+  // Ta-da! All that's left is to update the target input box:
+  targetForm.unit_input.value = result;
+}
+
+// This fragment initializes the property dropdown menu using the data defined above in the 'Data Definitions' section
+window.onload = function (e) {
+  FillMenuWithArray(document.property_form.the_menu, property);
+  UpdateUnitMenu(document.property_form.the_menu, document.form_A.unit_menu);
+  UpdateUnitMenu(document.property_form.the_menu, document.form_B.unit_menu)
+}
```

Comment:
I strongly suggest to use addEventListener as onload is commonly overwritten.
```suggestion
// This fragment initializes the property dropdown menu using the data defined above in the 'Data Definitions' section
window.addEventListener('load', function (e) {
  FillMenuWithArray(document.property_form.the_menu, property);
  UpdateUnitMenu(document.property_form.the_menu, document.form_A.unit_menu);
  UpdateUnitMenu(document.property_form.the_menu, document.form_B.unit_menu)
});
```